src/scip_data.jl

```diff
@@ -224,29 +224,13 @@ function add_quadratic_constraint(scipd::SCIPData, linrefs, lincoefs,
     quadvars2 = [var(scipd, vr) for vr in quadrefs2]
 
     cons__ = Ref{Ptr{SCIP_CONS}}(C_NULL)
-    @SCIP_CALL SCIPcreateConsBasicQuadratic(
+    @SCIP_CALL SCIPcreateConsBasicQuadraticNonlinear(
         scipd, cons__, "", length(linvars), linvars, lincoefs,
         length(quadvars1), quadvars1, quadvars2, quadcoefs, lhs, rhs)
     @SCIP_CALL SCIPaddCons(scipd, cons__[])
     return store_cons!(scipd, cons__)
 end
 
-"""
-Add second-order-cone constraint to problem, return cons ref.
-
-Does not support the full generality of SCIP's constraint (offsets and
-coefficients). The first entry in `varrefs` is used for the special variable on
-the right-hand side.
-"""
-function add_second_order_cone_constraint(scipd::SCIPData, varrefs)
-    vars = [var(scipd, vr) for vr in varrefs]
-    cons__ = Ref{Ptr{SCIP_CONS}}(C_NULL)
-    @SCIP_CALL SCIPcreateConsBasicSOC(scipd, cons__, "", length(vars) - 1,
-                               vars[2:end], C_NULL, C_NULL, 0.0, vars[1], 1.0, 0.0)
-    @SCIP_CALL SCIPaddCons(scipd, cons__[])
-    return store_cons!(scipd, cons__)
-end
-
```

Comment:
Is there no SOC constraint in SCIP 8, or will it simply be detected from the quadratic expression in the `cons_expr` handler?

Reply:
Similar for `cons_abspower`.

Reply:
it is detected from quadratic constraints. So we can add SOC in this way. Same thing for AbsPower. The thing that's still unclear to me is whether retrieving constraints is still doable in a specialized fashion or if, to retrieve constraint data, one has to get a nonlinear expression and recompute the quadratic/SOC/abspower data out of it

Reply:
There are no longer soc, abspower, or quadratic constraints in SCIP 8. It would be consistent if interfaces would not pretend that they still exist. There are nonlinear constraints and there are some convience methods to create nonlinear constraints with the data that was previously used to create quadratic, soc, or abspower constraints. But that is for creation, not really to get the data back in the same form. You can, however, ask whether a nonlinear constraint can be represented as quadratic or soc and get the corresponding data, but this detection assumes simplified expressions.

soc is detected from nonlinear constraints that have an expression that is quadratic (or sqrt(quadratic) in some cases).

Reply:
OK, thanks for the clarification. In that sense, it's right to remove the methods to create SOC, abspower etc.

Reply:
yes this would be a huge regression for the interface to remove all the structured constraints, SOCs and specialized constraints sure but quadratic constraints are fairly essential. 

> You can, however, ask whether a nonlinear constraint can be represented as quadratic or soc and get the corresponding data, but this detection assumes simplified expressions

then yes we should go this path

Reply:
@svigerske where can we look for these methods?